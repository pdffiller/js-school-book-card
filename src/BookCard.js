import React from 'react';

const BookAuthor = ({ author }) => (
  <p className="card-text">
    {author}
  </p>
);

const BookAgeBadge = ({ isNew }) => (
  <div className={`badge badge-${isNew ? 'success' : 'secondary'}`}>
    {isNew ? 'new' : 'old'}
  </div>
)

const BookCard = ({ thumbnail, title, authors, year }) => (
  <div className="col-md-4">
    <div className="card">
      <img
        className="card-img-top"
        src={thumbnail}
        alt={title}
      />
      <div className="card-body">
        <h5 className="card-title">{title}</h5>
          {authors ? authors.map(
            author => <BookAuthor key={author} author={author} />
          ) : 'no authors'}
        <div className="card-text">
          {year}
          {' '}
          <BookAgeBadge isNew={year > 2010} />
        </div>
      </div>
    </div>
  </div>
);

export default BookCard;
