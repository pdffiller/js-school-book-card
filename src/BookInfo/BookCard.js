import React from 'react';

const BookAuthor = ({ author }) => (
  <p className="card-text">
    {author}
  </p>
);

const BookAgeBadge = ({ isNew }) => (
  <div className={`badge badge-${isNew ? 'success' : 'secondary'}`}>
    {isNew ? 'new' : 'old'}
  </div>
)

const BookCard = ({ thumbnail, title, authors, year, toggleView }) => (
  <div className="col-md-4">
    <div className="card">
      <img
        className="card-img-top"
        src={thumbnail}
        alt={title}
      />
      <div className="card-body">
        <h4 className="card-title">
          <button onClick={toggleView} className="btn btn-link">
            {title}
          </button>
        </h4>
          {authors ? authors.map(
            author => <BookAuthor key={author} author={author} />
          ) : 'no authors'}
        <div className="card-text">
          {year}
          {' '}
          <BookAgeBadge isNew={year > 2010} />
        </div>
      </div>
    </div>
  </div>
);

export default BookCard;
