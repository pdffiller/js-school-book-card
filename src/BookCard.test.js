import React from 'react';
import renderer from 'react-test-renderer';
import BookCard from './BookCard';

describe('<BookCard>', () => {
  it('should renders without crashing', () => {
    const bookCardElement = renderer.create(
      <BookCard
        title="Happy"
        thumbnail="img.jpg"
        authors={['Dima']}
        year={2018}
      />
    );
  });

  it.skip('should renders without crashing if no props are passed', () => {
    const bookCardElement = renderer.create(
      <BookCard />
    );
  });

  it('should renders correctly for new books', () => {
    const bookCardElement = renderer.create(
      <BookCard
        title="Happy"
        thumbnail="img.jpg"
        authors={['Dima']}
        year={2018}
      />
    );

    expect(
      bookCardElement
    ).toMatchSnapshot();
  });

  it('should renders correctly for old books', () => {
    const bookCardElement = renderer.create(
      <BookCard
        title="Happy"
        thumbnail="img.jpg"
        authors={['Dima']}
        year={2000}
      />
    );

    expect(
      bookCardElement
    ).toMatchSnapshot();
  });
});