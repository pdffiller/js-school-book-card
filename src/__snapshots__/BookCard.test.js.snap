// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BookCard> should renders correctly for new books 1`] = `
<div
  className="col-md-4"
>
  <div
    className="card"
  >
    <img
      alt="Happy"
      className="card-img-top"
      src="img.jpg"
    />
    <div
      className="card-body"
    >
      <h5
        className="card-title"
      >
        Happy
      </h5>
      <p
        className="card-text"
      >
        Dima
      </p>
      <div
        className="card-text"
      >
        2018
         
        <div
          className="badge badge-success"
        >
          new
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<BookCard> should renders correctly for old books 1`] = `
<div
  className="col-md-4"
>
  <div
    className="card"
  >
    <img
      alt="Happy"
      className="card-img-top"
      src="img.jpg"
    />
    <div
      className="card-body"
    >
      <h5
        className="card-title"
      >
        Happy
      </h5>
      <p
        className="card-text"
      >
        Dima
      </p>
      <div
        className="card-text"
      >
        2000
         
        <div
          className="badge badge-secondary"
        >
          old
        </div>
      </div>
    </div>
  </div>
</div>
`;
